server/src/task/task.controller.ts

```diff
@@ -0,0 +1,60 @@
+import { AuthGuard } from '../auth/guard';
+import { CreateTaskDto, UpdateTaskDto, UpdateTaskStatusDto } from './dto';
+
+import { TaskService } from './task.service';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+
+@UseGuards(AuthGuard)
+@Controller('tasks')
+export class TaskController {
+  constructor(private taskService: TaskService) {}
+
+  // Create task
+  @Post('create')
+  createTask(@Body() createTaskDto: CreateTaskDto) {
+    return this.taskService.createTask(createTaskDto);
+  }
+
+  // Get task
+  @Get(':taskId')
+  GetTask(@Param('taskId', ParseIntPipe) taskId: number) {
+    return this.taskService.getTask(taskId);
+  }
+
+  // Edit task
+  @Patch(':taskId')
```

Comment:
Dto seems to be not optional so Put is correct?

Reply:
Fixed it!
https://github.com/HAK2024/fairy-share/pull/69/commits/d37ebc42b989efd92201ced4d9dd4fc8bc52fb3a